src/vmm/src/config/builder.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Config builder
+use std::convert::TryFrom;
+
+use super::{
+    BlockConfig, ConversionError, KernelConfig, MemoryConfig, NetConfig, VMMConfig, VcpuConfig,
+};
+
+/// Builder structure for VMMConfig
+#[derive(Debug)]
+pub struct Builder {
+    inner: Result<VMMConfig, ConversionError>,
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Builder {
+            inner: Ok(VMMConfig::default()),
+        }
+    }
+}
+
+impl Builder {
+    /// Creates a Builder Object
+    ///
+    pub fn new() -> Self {
+        Builder::default()
+    }
+
+    /// Builds VMMConfig
+    ///
+    /// This function should be last of the chain to build VMMConfiguration. Since, it is not known
+    /// which of the `config` functions user has called, missing configuration errors are handled
+    /// here -
+    ///  * Empty Kernel Configuration ('path' missing.)
+    ///
+    /// # Example
+    ///
+    /// ```
+    ///  # use vmm::VMMConfig;
+    ///
+    ///  let vmmconfig = VMMConfig::builder()
+    ///     .kernel_config(Some("path=/path/to/bzImage"))
+    ///     .build();
+    ///
+    /// assert!(vmmconfig.is_ok());
+    ///
+    /// ```
+    ///
+    pub fn build(&self) -> Result<VMMConfig, ConversionError> {
+        // Check if there are any errors
+        match &self.inner {
+            Ok(vc) => {
+                // Empty kernel image path.
+                if vc.kernel_config.path.to_str().unwrap().is_empty() {
+                    return Err(ConversionError::ParseKernel(
+                        "Kernel Image Path is Empty.".to_string(),
+                    ));
+                }
+            }
+            Err(_) => {}
+        }
+
+        self.inner.clone()
+    }
+
+    /// Configure Builder with Memory Configuration for the VMM
+    ///
+    /// # Example
```

Comment:
Just a quick comment, if the examples turn out to be very similar (as it happens in this module), I would suggest to have a single one at the module level that uses all the public functions, and in the `Example` section of each function to point to that one.

You can find a hopefully decent example with this approach in [this module](https://github.com/rust-vmm/vm-superio/blob/master/src/i8042.rs).